guardrails/integrations/llama_index/guardrails_query_engine.py

```diff
@@ -0,0 +1,122 @@
+from typing import Any, List, Optional, Dict, TYPE_CHECKING
+import asyncio
+import importlib
+
+from guardrails import Guard
+from guardrails.errors import ValidationError
+
+if TYPE_CHECKING:
+    from llama_index.core.query_engine import BaseQueryEngine
+    from llama_index.core.schema import QueryBundle, QueryType, NodeWithScore
+    from llama_index.core.callbacks import CallbackManager
+    from llama_index.core.base.response.schema import RESPONSE_TYPE
+else:
+    BaseQueryEngine = QueryBundle = NodeWithScore = CallbackManager = QueryType = (
+        RESPONSE_TYPE
+    ) = Any
+
+
+class GuardrailsQueryEngine:
+    def __init__(
+        self,
+        query_engine: "BaseQueryEngine",
+        input_guard: Optional[Guard] = None,
+        output_guard: Optional[Guard] = None,
+        input_guard_kwargs: Optional[Dict[str, Any]] = None,
+        output_guard_kwargs: Optional[Dict[str, Any]] = None,
+        callback_manager: Optional["CallbackManager"] = None,
+    ):
+        LLAMA_INDEX_CORE = importlib.util.find_spec("llama_index.core")
+        if LLAMA_INDEX_CORE is not None:
+            raise ImportError(
+                "llama_index is not installed. Please install it with "
+                "`pip install llama-index` to use GuardrailsQueryEngine."
+            )
+        self._query_engine = query_engine
+        self._input_guard = input_guard
+        self._output_guard = output_guard
+        self._input_guard_kwargs = input_guard_kwargs or {}
+        self._output_guard_kwargs = output_guard_kwargs or {}
+        self._callback_manager = callback_manager or CallbackManager([])
+
+    async def _validate_input(self, query: str) -> str:
+        if self._input_guard:
+            try:
+                async with self._callback_manager.event(event_type="input_validation"):
+                    validated_input = await self._input_guard.validate(
```

Comment:
If this is intended to be async by default then the input and output guards should be `AsyncGuard`'s